Frontend/src/screens/registerScreen.js

```diff
@@ -84,129 +87,109 @@ const RegisterScreen = ({navigation}) => {
-          placeholder="Enter Phone"
-          placeholderTextColor="#8b9cb5"
-          returnKeyType="next"
-          blurOnSubmit={false}
-        />
-      </View>
-      <View>
-        <Text>Address</Text>
-        <TextInput
-          onChangeText={(UserAddress) => setUserAddress(UserAddress)}
-          placeholder="Enter Address"
-          placeholderTextColor="#8b9cb5"
-          returnKeyType="next"
-          blurOnSubmit={false}
-        />
-      </View>
-      <View>
-        <Text>Birthday</Text>
-        <TextInput
-          onChangeText={(UserBirthday) => setUserBirthday(UserBirthday)}
-          placeholder="Enter Birthday"
-          placeholderTextColor="#8b9cb5"
-          returnKeyType="next"
-          blurOnSubmit={false}
-        />
-      </View>
-      <View>
-        <Text>Gender</Text>
-        <TextInput
-          onChangeText={(UserGender) => setUserGender(UserGender)}
-          placeholder="Enter Gender"
-          placeholderTextColor="#8b9cb5"
-          returnKeyType="next"
-          blurOnSubmit={false}
-        />
-      </View>
-      <View>
-        <Text>First Name</Text>
-        <TextInput
-          onChangeText={(UserFirstName) => setUserFirstName(UserFirstName)}
-          placeholder="Enter First Name"
-          placeholderTextColor="#8b9cb5"
-          returnKeyType="next"
-          blurOnSubmit={false}
-        />
-      </View>
-      <View>
-        <Text>Last Name</Text>
-        <TextInput
-          onChangeText={(UserLastName) => setUserLastName(UserLastName)}
-          placeholder="Enter Last Name"
-          placeholderTextColor="#8b9cb5"
-          returnKeyType="next"
-          blurOnSubmit={false}
-        />
-      </View>
-      <TouchableOpacity activeOpacity={0.5} onPress={handleSubmitButton}>
-        <Text>Sign up</Text>
-      </TouchableOpacity>
-      <View style={{flexDirection: 'row'}}>
-        <Text> Already have an account? </Text>
-        <Text onPress={() => navigation.navigate('LoginScreen')}>Sign In</Text>
-      </View>
-    </ScrollView>
+    <View style={styles.container}>
+      <ImageBackground
+        source={require('../images/authBackground.jpg')}
+        style={styles.backgroundImage}>
+        <View style={styles.header}>
+          <Text style={styles.headerText}>Welcome to the Gym Application</Text>
+        </View>
+        <View style={styles.body}>
+          <ScrollView>
+            <AuthInputField title="Email" setData={setUserEmail} />
+            <AuthInputField title="Username" setData={setUserName} />
+            <AuthInputField
+              title="Password"
+              setData={setUserPassword}
+              secureTextEntry={true}
+            />
+            <AuthInputField
+              title="Re-type Password"
+              setData={setUserRetypePassword}
+              secureTextEntry={true}
+            />
+            <AuthInputField title="Phone" setData={setUserPhone} />
```

Comment:
Maybe split all these fields in two different screens or if you want them in the same screen at least two different components, the render methods becomes too big.